src/coreclr/jit/optimizer.cpp

```diff
@@ -6079,7 +6068,7 @@ void Compiler::optRemoveRedundantZeroInits()
                             // insert a call to CORINFO_HELP_INIT_PINVOKE_FRAME but that is not a gc-safe point.
                             assert(emitter::emitNoGChelper(CORINFO_HELP_INIT_PINVOKE_FRAME));
 
-                            if (!lclDsc->HasGCPtr() || (!GetInterruptible() && !hasGCSafePoint))
+                            if (!lclDsc->HasGCPtr() || (!GetInterruptible() && !IsPotentialGCSafePoint(tree)))
```

Comment:
Should the `IsPotentialGCSafePoint` be called on all trees between prolog and the first use of the local and set `hasGCSafePoint` if there is an intervening safe point that may observe unassigned state?

Reply:
Oops, my bad, thanks for noticing! Fixed